kuma/javascript/src/payments/subscription-form.jsx

```diff
@@ -1,76 +1,181 @@
+                // token is only called if Stripe was able to successfully
+                // create a token from the entered info
+                token(response) {
+                    token.current = response.id;
+                    createSubscription();
                 },
-                closed: function() {
-                    if (!formData.get('stripe_token')) {
-                        setIsSubmitting(false);
-                    }
+                closed() {
+                    setFormStep(token.current ? 'sent' : 'initial');
                 }
             });
-
             stripeHandler.open();
-        }
-    };
+        });
+    }
 
-    return (
-        <div className="subscriptions-form">
-            <header className="subscriptions-form-header">
-                <h2>
-                    <Interpolated
-                        id={gettext('$5 <perMontSub />')}
-                        perMontSub={<sub>{gettext('/mo')}</sub>}
-                    />
-                </h2>
-            </header>
+    function createSubscription() {
+        fetch(SUBSCRIPTION_URL, {
+            method: 'POST',
+            body: JSON.stringify({
+                stripe_token: token.current // eslint-disable-line camelcase
+            }),
+            headers: {
+                'X-CSRFToken': getCookie('csrftoken'),
+                'Content-Type': 'application/json'
+            }
+        })
+            .then(() => {
+                window.location = `/${locale}/payments/thank-you/`;
+            })
+            .catch(e => {
+                console.error('error while creating subscription', e);
+                setFormStep('server_error');
+            });
+    }
+
+    let content;
+    if (formStep === 'server_error') {
+        content = (
+            <>
+                <h2>{gettext('Sorry!')}</h2>
+                <p>
+                    {gettext(
+                        "An error occurred trying to set up the subscription with Stripe's server. We've recorded the error and will investigate it."
+                    )}
+                </p>
+                <button
+                    type="button"
+                    className="button cta primary"
+                    onClick={() => setFormStep('initial')}
+                >
+                    {gettext('Try again')}
+                </button>
+            </>
+        );
+    } else if (formStep === 'stripe_error') {
+        content = (
+            <>
+                <h2>{gettext('Sorry!')}</h2>
+                <p>
+                    {gettext(
+                        'An error happened trying to load the Stripe integration'
+                    )}
+                </p>
+                <button
+                    type="button"
+                    className="button cta primary"
+                    onClick={reloadStripe}
+                >
+                    {gettext('Try again')}
+                </button>
+            </>
```

Comment:
Do you think you can address the styling a little. Here's what happens when the stripe SDK can't be loaded:
<img width="1616" alt="Screen Shot 2020-03-30 at 3 37 33 PM" src="https://user-images.githubusercontent.com/26739/77954478-a6601200-729c-11ea-8bd7-1d6a787f23d9.png">